File: Chapter 13/C#/Section_13_3_2_ArchitectureGovernance/ShopZilla.OrderService/Domain/OrderStatus.cs
namespace ShopZilla.OrderService.Domain
{
    /// <summary>
    /// The lifecycle states of an Order.
    ///
    /// This enum lives in the Domain layer alongside the Order entity.
    /// It represents pure business vocabulary with no infrastructure
    /// dependencies whatsoever.
    /// </summary>
    public enum OrderStatus
    {
        /// <summary>Order has been created but not yet paid.</summary>
        Pending = 0,

        /// <summary>Payment has been received.</summary>
        Paid = 1,

        /// <summary>Order has been dispatched to the customer.</summary>
        Shipped = 2,

        /// <summary>Order was cancelled before shipping.</summary>
        Cancelled = 3
    }
}

File: Chapter 13/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter13/section_13_3_2_architecture_governance/domain/OrderStatus.java
package com.grokkingsoftwarearchitecture.chapter13.section_13_3_2_architecture_governance.domain;

/**
 * The lifecycle states of an Order.
 *
 * This enum lives in the Domain layer alongside the Order entity.
 * It represents pure business vocabulary with no infrastructure
 * dependencies whatsoever.
 */
public enum OrderStatus {
    /** Order has been created but not yet paid. */
    PENDING,

    /** Payment has been received. */
    PAID,

    /** Order has been dispatched to the customer. */
    SHIPPED,

    /** Order was cancelled before shipping. */
    CANCELLED
}